haskell/private/cabal_wrapper.sh.tpl

```diff
@@ -12,15 +12,27 @@
 
 set -euo pipefail
 shopt -s nullglob
-execroot="$(pwd)"
 
-# Set these variables if unset.
-: ${LD_LIBRARY_PATH:=}
-: ${LIBRARY_PATH:=}
-export LD_LIBRARY_PATH=$execroot/${LD_LIBRARY_PATH//:/:$execroot/}
-export LIBRARY_PATH=$execroot/${LIBRARY_PATH//:/:$execroot/}
+function canonicalize_path()
+{
+    new_path=""
+    while IFS=: read -r -d: entry
+    do
+	if [[ -n "$entry" ]]
+	then
+	    new_path="$new_path${new_path:+:}$(realpath -m "$entry")"
```

Comment:
As far as I can tell `realpath` doesn't exist on macos, so this will probably fail on non nixpkgs macos. This can also be done in bash, similar to [here](https://github.com/tweag/rules_haskell/blob/1f7a51b278f27b0cea8c9aef9b529e39ea735b75/hazel/third_party/cabal2bazel/bzl/cabal_package.bzl#L91)

Reply:
macOS+Nix shouldn't be a problem, but bare macOS, yes. A pure bash solution is better. I found this on the Internet:

```
realpath() {
    [[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"
}
```

which I understand better than the code you link to.

Reply:
The linked code does essentially the same thing, with the difference that it keeps relative paths relative. In that case it matters more, as the outputs of `./configure` are outputs of the configure step and inputs to later build steps, so absolute paths would certainly leak into cache keys. Here it should matter less, as `PATH` should typically not end up in the built library/binary.